### PR TITLE
migrate fork util to SDK

### PR DESCRIPTION
**Pull request by nbayindirli**
### Description

* Migrates fork util to SDK

### Drive-by changes

* None

### Related issues

- Fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3624

### Backward compatibility

* Yes

### Testing

* Manual on core & warp

### CODE REVIEW

File: typescript/sdk/src/utils/env.ts
Comment (jmrossy) [inline typescript/sdk/src/utils/env.ts:9]: accessing `process.foo` will throw an error when run in the browser, where `process` is undefined.
That's what the `safelyAccessEnvVar` utils is about. See logging.ts in utils for an example

File: typescript/sdk/src/utils/fork.ts
Comment (jmrossy) [inline typescript/sdk/src/utils/fork.ts:20]: I recommend making a child logger for this file. `rootLogger.child({module:'fork-utils'})`

File: typescript/sdk/src/utils/fork.ts
Comment (jmrossy) [inline typescript/sdk/src/utils/fork.ts:3]: If this is in the SDK it shouldn't import from the sdk package like this

**Review by jmrossy — CHANGES_REQUESTED** [review]
Needs a few fixes before we can merge :)

File: typescript/sdk/src/utils/env.ts
Comment (nbayindirli) [reply typescript/sdk/src/utils/env.ts:9]: Gotcha– did not have viz into this. Thanks for catching

File: typescript/sdk/src/utils/env.ts
Comment (jmrossy) [inline typescript/sdk/src/utils/env.ts:11]: Hmm, sorry if I wasn't clear before but I don't think this is quite right either. Wouldn't this try to access an env var called `env`? so `process.env['env']`. Was that the intention or are each of the env vars above meant to be set individually?

**Review by yorhodes — APPROVED** [review]

File: typescript/sdk/src/utils/env.ts
Comment (nbayindirli) [reply typescript/sdk/src/utils/env.ts:9]: Agreed– I've now removed the HYP_KEY from sdk-side, and similarly the ANVIL vars from the CLI-side.

Are you also against ANVIL vars sdk-side? We can certainly have these be passed into the fork util instead of parsed out of process.env

Comment (nbayindirli) [reply typescript/sdk/src/utils/env.ts:9]: ^talking through it, I think that's my desired path now as well

File: typescript/sdk/src/utils/env.ts
Comment (nbayindirli) [reply typescript/sdk/src/utils/env.ts:11]: Continuing– it now seems appropriate to encapsulate fork.ts in a class, given that it lives in SDK side + enables IO from CLI, which may need to persist certain values (recently discussed [here](https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3586#discussion_r1564069199). By doing so, we can expose IP/port setters, which can eventually enable multi-chain support (when multiple Anvil instances will need to run)

cc @yorhodes

Comment (nbayindirli) [reply typescript/sdk/src/utils/env.ts:11]: Just to keep this PR moving, I've captured work to do this here: https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3652

Will hold off on adding this to the current change

**Review by jmrossy — APPROVED** [review]

**Pull request merged**
